api/http/proxy/factory_linux.go

```diff
@@ -0,0 +1,12 @@
+// +build linux
+
+package proxy
+
+import (
+	"net/http"
+)
+
+func (factory *proxyFactory) newNamedPipeProxy(path string) http.Handler {
```

Comment:
This function should be renamed `newLocalProxy` and it's content should be the actual content of `newDockerSocketProxy` (relocate the function here and rename it).